src/autobot/internal/draw/abpaintprovider.cpp

```diff
@@ -38,14 +39,15 @@ void AbPaintProvider::beginTarget(const std::string& name)
     BufferedPaintProvider::beginTarget(name);
 }
 
-bool AbPaintProvider::endTarget(const std::string& name, bool endDraw)
+bool AbPaintProvider::endTarget(bool endDraw)
 {
-    UNUSED(endDraw);
-    LOGI() << name;
-    return BufferedPaintProvider::endTarget(name, endDraw);
+    bool ok = BufferedPaintProvider::endTarget(endDraw);
+    if (ok && buffer().name == "notationview") {
```

Comment:
Please explain why you need this check?

Reply:
In normal, one paint provider instance for one painter.
In this test, one paint provider instance for all painters (that is, in fact, it is a singleton). This name contains what is currently being painted (there may also be palettes, export, etc.)
In this case, we need only notation view painting